tests/cases/conformance/classes/propertyMemberDeclarations/assignParameterPropertyToPropertyDeclarationESNext.ts

```diff
@@ -0,0 +1,17 @@
+// @useDefineForClassFields: true
+// @target: esnext
+class C {
+    qux = this.bar // should error
+    bar = this.foo // should error
+    quiz = this.bar // ok
```

Comment:
Given the code change, I don't understand why the lines with `// ok` don't error. It is *nice* that they don't though.